### ISSUE TITLE
Cannot open order on backoffice with deleted products

### ISSUE THREAD

**Issue by kittysgifts** [PrestaShop/PrestaShop#35753]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

    Run Prestashop 8.1.3;
    Delete a product in a order;
    Try to see the order detail.

We'll get an error:

"PrestaShop\PrestaShop\Core\Domain\Order\QueryResult\OrderProductForViewing::__construct(): Argument #19 ($location) must be of type string, null given, called in public_html/src/Adapter/Order/QueryHandler/GetOrderProductsForViewingHandler.php on line 279"

Specs

When a product is deleted, we cannot open the order detail anymore. I got this error in old orders, but since I had a new one closed, I’ve deleted one of the products that didn’t have stock anymore, to see if it happens with this new order.

Everything was ok until I delete the product (I’ve seen the order detail several times). Now I cannot open the detail.
I've investigated what I could and the error is about "location" field  - I've thought since the table entry of the product was removed from ps_product - now, in the select this field is getting the "null" value, that leads to this error.

I cannot investigate anymore since I don't know where to find the select and I'm not familiarized with PHP.

Additional information

    PrestaShop version: 8.1.3
    PHP version: 8.1.27

### Expected behavior

We should be able to open the order details even when it has deleted products.

### Steps to reproduce

    Run Prestashop 8.1.3;
    Delete a product in a order;
    Try to see the order detail.

We'll get an error:

"PrestaShop\PrestaShop\Core\Domain\Order\QueryResult\OrderProductForViewing::__construct(): Argument #19 ($location) must be of type string, null given, called in public_html/src/Adapter/Order/QueryHandler/GetOrderProductsForViewingHandler.php on line 279"
![erro](https://github.com/PrestaShop/PrestaShop/assets/77332737/9827d3b4-76de-4dda-851e-e55721f58103)

### PrestaShop version(s) where the bug happened

1.8.3

### PHP version(s) where the bug happened

8.1.27

**Comment by Hlavtox** [comment]
Duplicate of https://github.com/PrestaShop/PrestaShop/issues/34454

Run this to fix your install. If the error comes back, we can investigate. 👍

```sql
UPDATE `ps_address` SET `address2`='' WHERE `address2` IS NULL;
UPDATE `ps_address` SET `company`='' WHERE `company` IS NULL;
UPDATE `ps_address` SET `dni`='' WHERE `dni` IS NULL;
UPDATE `ps_address` SET `id_state`='0' WHERE `id_state` IS NULL;
UPDATE `ps_address` SET `other`='' WHERE `other` IS NULL;
UPDATE `ps_address` SET `phone_mobile`='' WHERE `phone_mobile` IS NULL;
UPDATE `ps_address` SET `phone`='' WHERE `phone` IS NULL;
UPDATE `ps_address` SET `postcode`='' WHERE `postcode` IS NULL;
UPDATE `ps_address` SET `vat_number`='' WHERE `vat_number` IS NULL;
UPDATE `ps_attachment_lang` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_carrier_lang` SET `delay`='' WHERE `delay` IS NULL;
UPDATE `ps_carrier` SET `external_module_name`='' WHERE `external_module_name` IS NULL;
UPDATE `ps_carrier` SET `url`='' WHERE `url` IS NULL;
UPDATE `ps_cart_rule` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_cart` SET `gift_message`='' WHERE `gift_message` IS NULL;
UPDATE `ps_category_lang` SET `additional_description`='' WHERE `additional_description` IS NULL;
UPDATE `ps_category_lang` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_category_lang` SET `meta_description`='' WHERE `meta_description` IS NULL;
UPDATE `ps_category_lang` SET `meta_keywords`='' WHERE `meta_keywords` IS NULL;
UPDATE `ps_category_lang` SET `meta_title`='' WHERE `meta_title` IS NULL;
UPDATE `ps_cms_category_lang` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_cms_category_lang` SET `meta_description`='' WHERE `meta_description` IS NULL;
UPDATE `ps_cms_category_lang` SET `meta_keywords`='' WHERE `meta_keywords` IS NULL;
UPDATE `ps_cms_category_lang` SET `meta_title`='' WHERE `meta_title` IS NULL;
UPDATE `ps_cms_lang` SET `content`='' WHERE `content` IS NULL;
UPDATE `ps_cms_lang` SET `head_seo_title`='' WHERE `head_seo_title` IS NULL;
UPDATE `ps_cms_lang` SET `meta_description`='' WHERE `meta_description` IS NULL;
UPDATE `ps_cms_lang` SET `meta_keywords`='' WHERE `meta_keywords` IS NULL;
UPDATE `ps_configuration_kpi_lang` SET `date_upd`=CURRENT_TIMESTAMP WHERE `date_upd` IS NULL;
UPDATE `ps_configuration_kpi_lang` SET `value`='' WHERE `value` IS NULL;
UPDATE `ps_configuration_lang` SET `date_upd`=CURRENT_TIMESTAMP WHERE `date_upd` IS NULL;
UPDATE `ps_configuration_lang` SET `value`='' WHERE `value` IS NULL;
UPDATE `ps_connections_source` SET `http_referer`='' WHERE `http_referer` IS NULL;
UPDATE `ps_connections_source` SET `keywords`='' WHERE `keywords` IS NULL;
UPDATE `ps_connections` SET `http_referer`='' WHERE `http_referer` IS NULL;
UPDATE `ps_connections` SET `ip_address`='0' WHERE `ip_address` IS NULL;
UPDATE `ps_contact_lang` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_currency_lang` SET `pattern`='' WHERE `pattern` IS NULL;
UPDATE `ps_customer_message` SET `file_name`='' WHERE `file_name` IS NULL;
UPDATE `ps_customer_message` SET `id_employee`='0' WHERE `id_employee` IS NULL;
UPDATE `ps_customer_message` SET `ip_address`='' WHERE `ip_address` IS NULL;
UPDATE `ps_customer_message` SET `user_agent`='' WHERE `user_agent` IS NULL;
UPDATE `ps_customer_thread` SET `id_order`='0' WHERE `id_order` IS NULL;
UPDATE `ps_customer_thread` SET `id_product`='0' WHERE `id_product` IS NULL;
UPDATE `ps_customer` SET `birthday`='0000-00-00' WHERE `birthday` IS NULL;
UPDATE `ps_customer` SET `newsletter_date_add`='0000-00-00 00:00:00' WHERE `newsletter_date_add` IS NULL;
UPDATE `ps_customer` SET `reset_password_validity`='0000-00-00 00:00:00' WHERE `reset_password_validity` IS NULL;
UPDATE `ps_employee` SET `bo_css`='' WHERE `bo_css` IS NULL;
UPDATE `ps_employee` SET `reset_password_validity`='0000-00-00 00:00:00' WHERE `reset_password_validity` IS NULL;
UPDATE `ps_employee` SET `stats_compare_from`='0000-00-00' WHERE `stats_compare_from` IS NULL;
UPDATE `ps_employee` SET `stats_compare_to`='0000-00-00' WHERE `stats_compare_to` IS NULL;
UPDATE `ps_employee` SET `stats_date_from`=CURRENT_TIMESTAMP WHERE `stats_date_from` IS NULL;
UPDATE `ps_employee` SET `stats_date_to`=CURRENT_TIMESTAMP WHERE `stats_date_to` IS NULL;
UPDATE `ps_feature_value` SET `custom`='0' WHERE `custom` IS NULL;
UPDATE `ps_guest` SET `accept_language`='' WHERE `accept_language` IS NULL;
UPDATE `ps_guest` SET `adobe_director`='0' WHERE `adobe_director` IS NULL;
UPDATE `ps_guest` SET `adobe_flash`='0' WHERE `adobe_flash` IS NULL;
UPDATE `ps_guest` SET `apple_quicktime`='0' WHERE `apple_quicktime` IS NULL;
UPDATE `ps_guest` SET `id_customer`='0' WHERE `id_customer` IS NULL;
UPDATE `ps_guest` SET `id_operating_system`='0' WHERE `id_operating_system` IS NULL;
UPDATE `ps_guest` SET `id_web_browser`='0' WHERE `id_web_browser` IS NULL;
UPDATE `ps_guest` SET `real_player`='0' WHERE `real_player` IS NULL;
UPDATE `ps_guest` SET `screen_color`='0' WHERE `screen_color` IS NULL;
UPDATE `ps_guest` SET `screen_resolution_x`='0' WHERE `screen_resolution_x` IS NULL;
UPDATE `ps_guest` SET `screen_resolution_y`='0' WHERE `screen_resolution_y` IS NULL;
UPDATE `ps_guest` SET `sun_java`='0' WHERE `sun_java` IS NULL;
UPDATE `ps_guest` SET `windows_media`='0' WHERE `windows_media` IS NULL;
UPDATE `ps_hook` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_image_lang` SET `legend`='' WHERE `legend` IS NULL;
UPDATE `ps_log` SET `error_code`='0' WHERE `error_code` IS NULL;
UPDATE `ps_log` SET `id_employee`='0' WHERE `id_employee` IS NULL;
UPDATE `ps_log` SET `id_lang`='0' WHERE `id_lang` IS NULL;
UPDATE `ps_log` SET `object_id`='0' WHERE `object_id` IS NULL;
UPDATE `ps_log` SET `object_type`='' WHERE `object_type` IS NULL;
UPDATE `ps_manufacturer_lang` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_manufacturer_lang` SET `meta_description`='' WHERE `meta_description` IS NULL;
UPDATE `ps_manufacturer_lang` SET `meta_keywords`='' WHERE `meta_keywords` IS NULL;
UPDATE `ps_manufacturer_lang` SET `meta_title`='' WHERE `meta_title` IS NULL;
UPDATE `ps_manufacturer_lang` SET `short_description`='' WHERE `short_description` IS NULL;
UPDATE `ps_message` SET `id_employee`='0' WHERE `id_employee` IS NULL;
UPDATE `ps_meta_lang` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_meta_lang` SET `keywords`='' WHERE `keywords` IS NULL;
UPDATE `ps_meta_lang` SET `title`='' WHERE `title` IS NULL;
UPDATE `ps_order_carrier` SET `id_order_invoice`='0' WHERE `id_order_invoice` IS NULL;
UPDATE `ps_order_carrier` SET `shipping_cost_tax_excl`='0' WHERE `shipping_cost_tax_excl` IS NULL;
UPDATE `ps_order_carrier` SET `shipping_cost_tax_incl`='0' WHERE `shipping_cost_tax_incl` IS NULL;
UPDATE `ps_order_carrier` SET `tracking_number`='' WHERE `tracking_number` IS NULL;
UPDATE `ps_order_carrier` SET `weight`='0' WHERE `weight` IS NULL;
UPDATE `ps_order_detail` SET `download_deadline`='0000-00-00 00:00:00' WHERE `download_deadline` IS NULL;
UPDATE `ps_order_detail` SET `download_hash`='' WHERE `download_hash` IS NULL;
UPDATE `ps_order_detail` SET `id_order_invoice`='0' WHERE `id_order_invoice` IS NULL;
UPDATE `ps_order_detail` SET `product_attribute_id`='0' WHERE `product_attribute_id` IS NULL;
UPDATE `ps_order_detail` SET `product_ean13`='' WHERE `product_ean13` IS NULL;
UPDATE `ps_order_detail` SET `product_isbn`='' WHERE `product_isbn` IS NULL;
UPDATE `ps_order_detail` SET `product_mpn`='' WHERE `product_mpn` IS NULL;
UPDATE `ps_order_detail` SET `product_reference`='' WHERE `product_reference` IS NULL;
UPDATE `ps_order_detail` SET `product_supplier_reference`='' WHERE `product_supplier_reference` IS NULL;
UPDATE `ps_order_detail` SET `product_upc`='' WHERE `product_upc` IS NULL;
UPDATE `ps_order_invoice` SET `delivery_date`='0000-00-00 00:00:00' WHERE `delivery_date` IS NULL;
UPDATE `ps_order_invoice` SET `note`='' WHERE `note` IS NULL;
UPDATE `ps_order_invoice` SET `shop_address`='' WHERE `shop_address` IS NULL;
UPDATE `ps_order_payment` SET `card_brand`='' WHERE `card_brand` IS NULL;
UPDATE `ps_order_payment` SET `card_expiration`='' WHERE `card_expiration` IS NULL;
UPDATE `ps_order_payment` SET `card_holder`='' WHERE `card_holder` IS NULL;
UPDATE `ps_order_payment` SET `card_number`='' WHERE `card_number` IS NULL;
UPDATE `ps_order_payment` SET `transaction_id`='' WHERE `transaction_id` IS NULL;
UPDATE `ps_order_return_state` SET `color`='' WHERE `color` IS NULL;
UPDATE `ps_order_slip_detail` SET `amount_tax_excl`='0' WHERE `amount_tax_excl` IS NULL;
UPDATE `ps_order_slip_detail` SET `amount_tax_incl`='0' WHERE `amount_tax_incl` IS NULL;
UPDATE `ps_order_slip_detail` SET `total_price_tax_excl`='0' WHERE `total_price_tax_excl` IS NULL;
UPDATE `ps_order_slip_detail` SET `total_price_tax_incl`='0' WHERE `total_price_tax_incl` IS NULL;
UPDATE `ps_order_slip_detail` SET `unit_price_tax_excl`='0' WHERE `unit_price_tax_excl` IS NULL;
UPDATE `ps_order_slip_detail` SET `unit_price_tax_incl`='0' WHERE `unit_price_tax_incl` IS NULL;
UPDATE `ps_order_slip` SET `total_products_tax_excl`='0' WHERE `total_products_tax_excl` IS NULL;
UPDATE `ps_order_slip` SET `total_products_tax_incl`='0' WHERE `total_products_tax_incl` IS NULL;
UPDATE `ps_order_slip` SET `total_shipping_tax_excl`='0' WHERE `total_shipping_tax_excl` IS NULL;
UPDATE `ps_order_slip` SET `total_shipping_tax_incl`='0' WHERE `total_shipping_tax_incl` IS NULL;
UPDATE `ps_order_state` SET `color`='' WHERE `color` IS NULL;
UPDATE `ps_order_state` SET `module_name`='' WHERE `module_name` IS NULL;
UPDATE `ps_orders` SET `gift_message`='' WHERE `gift_message` IS NULL;
UPDATE `ps_orders` SET `note`='' WHERE `note` IS NULL;
UPDATE `ps_product_attribute_lang` SET `available_later`='' WHERE `available_later` IS NULL;
UPDATE `ps_product_attribute_lang` SET `available_now`='' WHERE `available_now` IS NULL;
UPDATE `ps_product_attribute_shop` SET `available_date`='0000-00-00' WHERE `available_date` IS NULL;
UPDATE `ps_product_attribute_shop` SET `low_stock_threshold`='0' WHERE `low_stock_threshold` IS NULL;
UPDATE `ps_product_attribute` SET `available_date`='0000-00-00' WHERE `available_date` IS NULL;
UPDATE `ps_product_attribute` SET `ean13`='' WHERE `ean13` IS NULL;
UPDATE `ps_product_attribute` SET `isbn`='' WHERE `isbn` IS NULL;
UPDATE `ps_product_attribute` SET `low_stock_threshold`='0' WHERE `low_stock_threshold` IS NULL;
UPDATE `ps_product_attribute` SET `mpn`='' WHERE `mpn` IS NULL;
UPDATE `ps_product_attribute` SET `reference`='' WHERE `reference` IS NULL;
UPDATE `ps_product_attribute` SET `supplier_reference`='' WHERE `supplier_reference` IS NULL;
UPDATE `ps_product_attribute` SET `upc`='' WHERE `upc` IS NULL;
UPDATE `ps_product_download` SET `date_expiration`='0000-00-00 00:00:00' WHERE `date_expiration` IS NULL;
UPDATE `ps_product_download` SET `nb_days_accessible`='0' WHERE `nb_days_accessible` IS NULL;
UPDATE `ps_product_lang` SET `available_later`='' WHERE `available_later` IS NULL;
UPDATE `ps_product_lang` SET `available_now`='' WHERE `available_now` IS NULL;
UPDATE `ps_product_lang` SET `delivery_in_stock`='' WHERE `delivery_in_stock` IS NULL;
UPDATE `ps_product_lang` SET `delivery_out_stock`='' WHERE `delivery_out_stock` IS NULL;
UPDATE `ps_product_lang` SET `description_short`='' WHERE `description_short` IS NULL;
UPDATE `ps_product_lang` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_product_lang` SET `meta_description`='' WHERE `meta_description` IS NULL;
UPDATE `ps_product_lang` SET `meta_keywords`='' WHERE `meta_keywords` IS NULL;
UPDATE `ps_product_lang` SET `meta_title`='' WHERE `meta_title` IS NULL;
UPDATE `ps_product_sale` SET `date_upd`=CURRENT_TIMESTAMP WHERE `date_upd` IS NULL;
UPDATE `ps_product_shop` SET `available_date`='0000-00-00' WHERE `available_date` IS NULL;
UPDATE `ps_product_shop` SET `cache_default_attribute`='0' WHERE `cache_default_attribute` IS NULL;
UPDATE `ps_product_shop` SET `id_category_default`='0' WHERE `id_category_default` IS NULL;
UPDATE `ps_product_shop` SET `low_stock_threshold`='0' WHERE `low_stock_threshold` IS NULL;
UPDATE `ps_product_shop` SET `unity`='' WHERE `unity` IS NULL;
UPDATE `ps_product_supplier` SET `product_supplier_reference`='' WHERE `product_supplier_reference` IS NULL;
UPDATE `ps_product` SET `available_date`='0000-00-00' WHERE `available_date` IS NULL;
UPDATE `ps_product` SET `cache_default_attribute`='0' WHERE `cache_default_attribute` IS NULL;
UPDATE `ps_product` SET `ean13`='' WHERE `ean13` IS NULL;
UPDATE `ps_product` SET `id_category_default`='0' WHERE `id_category_default` IS NULL;
UPDATE `ps_product` SET `id_manufacturer`='0' WHERE `id_manufacturer` IS NULL;
UPDATE `ps_product` SET `id_supplier`='0' WHERE `id_supplier` IS NULL;
UPDATE `ps_product` SET `isbn`='' WHERE `isbn` IS NULL;
UPDATE `ps_product` SET `low_stock_threshold`='0' WHERE `low_stock_threshold` IS NULL;
UPDATE `ps_product` SET `mpn`='' WHERE `mpn` IS NULL;
UPDATE `ps_product` SET `reference`='' WHERE `reference` IS NULL;
UPDATE `ps_product` SET `supplier_reference`='' WHERE `supplier_reference` IS NULL;
UPDATE `ps_product` SET `unity`='' WHERE `unity` IS NULL;
UPDATE `ps_product` SET `upc`='' WHERE `upc` IS NULL;
UPDATE `ps_risk` SET `color`='' WHERE `color` IS NULL;
UPDATE `ps_stock` SET `ean13`='' WHERE `ean13` IS NULL;
UPDATE `ps_stock` SET `isbn`='' WHERE `isbn` IS NULL;
UPDATE `ps_stock` SET `mpn`='' WHERE `mpn` IS NULL;
UPDATE `ps_stock` SET `upc`='' WHERE `upc` IS NULL;
UPDATE `ps_store_lang` SET `address2`='' WHERE `address2` IS NULL;
UPDATE `ps_store_lang` SET `note`='' WHERE `note` IS NULL;
UPDATE `ps_store` SET `email`='' WHERE `email` IS NULL;
UPDATE `ps_store` SET `fax`='' WHERE `fax` IS NULL;
UPDATE `ps_store` SET `id_state`='0' WHERE `id_state` IS NULL;
UPDATE `ps_store` SET `phone`='' WHERE `phone` IS NULL;
UPDATE `ps_supplier_lang` SET `description`='' WHERE `description` IS NULL;
UPDATE `ps_supplier_lang` SET `meta_description`='' WHERE `meta_description` IS NULL;
UPDATE `ps_supplier_lang` SET `meta_keywords`='' WHERE `meta_keywords` IS NULL;
UPDATE `ps_supplier_lang` SET `meta_title`='' WHERE `meta_title` IS NULL;
UPDATE `ps_supply_order_detail` SET `ean13`='' WHERE `ean13` IS NULL;
UPDATE `ps_supply_order_detail` SET `isbn`='' WHERE `isbn` IS NULL;
UPDATE `ps_supply_order_detail` SET `mpn`='' WHERE `mpn` IS NULL;
UPDATE `ps_supply_order_detail` SET `upc`='' WHERE `upc` IS NULL;
UPDATE `ps_supply_order_state` SET `color`='' WHERE `color` IS NULL;
UPDATE `ps_supply_order` SET `date_delivery_expected`='0000-00-00 00:00:00' WHERE `date_delivery_expected` IS NULL;
UPDATE `ps_warehouse_product_location` SET `location`='' WHERE `location` IS NULL;
UPDATE `ps_warehouse` SET `reference`='' WHERE `reference` IS NULL;
UPDATE `ps_webservice_account` SET `description`='' WHERE `description` IS NULL;
```

**Comment by kittysgifts** [comment]
Thanks for you script. I've already run it but this problem it's different since that it solve data problems that exists on database. 

I've deleted the product - it doesn't exist anymore in the tables so, somewhere where it collects the data is putting the "location" field as null, which leads to the error.

It's all right in the preview - in yellow are the products that I deleted on this order:

![imagem](https://github.com/PrestaShop/PrestaShop/assets/77332737/04b5e1df-0135-46f3-9c2b-7ed5cdd504f8)

But in this preview it doesn't need the location (my guess), so there is no error.

When we open the order it shows the error:

![imagem](https://github.com/PrestaShop/PrestaShop/assets/77332737/2f1f39f3-68db-4e4a-a0bd-64c17ac0e0c9)

Thanks for you help.

**Comment by AureRita** [comment]
Hi @kittysgifts 

Thank you for your report, Desptie my effort, I can't reproduce your issue as you can see : 

[recording(160).webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/efd0d7ac-a08a-4d7f-a6c9-1c21eb4d8c5b)

Did I miss something ? 

FYI : to test issues, we need to test it on the last released version (8.1.x) 

Waiting for feedback

**Comment by kittysgifts** [comment]
Hi.

Thanks again for your help.

So, I was delaying the upgrade to 8.1.5, which I did today so I can be certain that the bug still happens.

I did it and I’ve tried to open the orders with delete products and it all works fine now. Whatever bug it has it seems to be corrected.

Thanks for your time and efforts.

Best Regards,

Sónia
